### ISSUE TITLE
Genshin since today - 100% CPU usage and new maybe new kernel-anticheat stuff

### ISSUE THREAD

**Issue by danilw** [an-anime-team/an-anime-game-launcher#383]
[I asked on reddit](https://www.reddit.com/r/linux_gaming/comments/1ddlfly/genshin_100_cpu_usage_since_today/) and it seems it confirmed.

**Since today - launching Genshin in Linux always use 100% CPU.**

*Current solution to 100% CPU usage in Genshin on Linux - turn off internet, start Game, wait for login screen and error, connect to internet - no 100% CPU.*

---

~~**Edit - Update 5.0** - seems no more "100% CPU" problem, offline launch not needed anymore.~~

**Edit - 5.0 - it seems 100% CPU is back** - and only solution still is offline launch.

---

**Edit - that log lines for anticheat is old, it was not working like that for long time.** 

*So maybe this is not reason to worry.*

---

Wine log, this last two lines:

```
0164:err:module:import_dll Library WDFLDR.SYS (which is needed by L"C:\\windows\\system32\\HoYoKProtect.sys") not found
0164:err:ntoskrnl:ZwLoadDriver failed to create driver L"\\Registry\\Machine\\System\\CurrentControlSet\\Services\\HoYoProtect": c0000142
```

After that - infinite spam of `fixme:sync:NtCreateTransaction` :

```
0188:fixme:sync:NtCreateTransaction (nil), 0x1aaff40, 0x789f90, {28ec8348-894c-ffc1-d289-c1ff15f39902}, (nil), 0x00000000, 0x00000000, 0x00000000, (nil), (nil) stub.
0188:fixme:sync:NtCreateTransaction (nil), 0x1aaff40, (nil), (null), (nil), 0x00000000, 0x00000000, 0x00000000, (nil), (nil) stub.
0188:fixme:sync:NtCreateTransaction (nil), 0x1aaff40, (nil), (null), (nil), 0x00000000, 0x00000000, 0x00000000, (nil), (nil) stub.
```

`WDFLDR.SYS` - is windows kernel-modules loader, and it seems Genshint trying to load new `C:\\windows\\system32\\HoYoKProtect.sys` - and it can not that result in infinite spam of error.

Idk how important it is and will it lead to account bans because we can not load this anticheat...
*(having 100% CPU usage is very annoying)*

**Comment by mrvictory1** [comment]
the reddit thread you linked has a solution and I can confirm it works
> turn off internet, start Game, wait for login screen and error, connect to internet - no 100% CPU

**Comment by nattuu** [comment]
Huh, now THAT'S why I've only been getting like 20 fps out of 60 lately. Also it might be consuming more RAM, but idk. Will update on any bans
Launching the usual way:
![Screenshot_2024-06-12_21-32-28](https://github.com/an-anime-team/an-anime-game-launcher/assets/39896488/2e80f554-6197-4a7d-9531-d86065fd19bc)
After trying the workaround out, same location, same settings:
![Screenshot_2024-06-12_21-35-35](https://github.com/an-anime-team/an-anime-game-launcher/assets/39896488/df74940d-689f-404a-b600-4233fb842a5f)
However, I must also point out that I recently downloaded the new Hoyoplay launcher on my Windoze installation (for kicks and giggles), which might have somehow altered Genshit's files that I share with Linux through a symlink, but I'm not too sure either, can't spare more 80 gigs for a single test. Would love to know if anyone's downloaded that shady crap too and encountered the same issue

**Comment by danilw** [comment]
> However, I must also point out that I recently downloaded the new Hoyoplay launcher

I downloaded/used it weeks before current 100% CPU behavior - I do not think it related.

**Comment by Sasikuttan2163** [comment]
> > However, I must also point out that I recently downloaded the new Hoyoplay launcher
> 
> I downloaded/used it weeks before current 100% CPU behavior - I do not think it related.

You're right, it's not. I noticed this after the ~4 KB or so update in-game we had yesterday. This was not due to the launcher.

**Comment by DvidPiDev** [comment]
I don't seem to notice a 100% CPU usage, in fact the game is running fine for me. Well, until I run out of space because the syslog eats the whole root partition.

**Comment by 5310** [comment]
On my Steam Deck, capped at 7W and medium resolution, 30fps around Fontaine, I now get 80–90% CPU usage. After using the "Wifi" trick, it drops to the 30–40% I used to get last week :smiling_face_with_tear: 

There's definitely a massive change, but I don't get frame drops yet. I certainly get a gigantic battery drop though! Gotta look out for the syslog too now... Hope this works out

**Comment by DvidPiDev** [comment]
I was thinking of using an rsyslog rule to prevent the launcher from logging anything. Though I'm not sure if that's a good idea.

**Comment by DvidPiDev** [comment]
Despite me adding rsyslog rules to prevent the launcher from logging anything, setting WINEDEBUG to -all and -richedit, the fixmes were still getting logged. 

Guess we'll just have to wait.

**Comment by danilw** [comment]
@DvidPiDev 
`export WINEDEBUG=-all`
to turn off wine-error spam you had in yours https://github.com/an-anime-team/an-anime-game-launcher/issues/384

**Comment by mrvictory1** [comment]
I just use this as launch parameters. Internet is cut systemwide but it works
nmcli n off ; %command% & sleep 15 ; nmcli n on

**Comment by DvidPiDev** [comment]
> `export WINEDEBUG=-all`
I already tried that, both in the .bashrc file (after sourcing it of course) and as launch parameters for the game.

All to no avail

**Comment by DvidPiDev** [comment]
Oh! It seems as though I had to fully restart my system for it to actually apply.
The syslog thing is solved then, thanks.

**Comment by danilw** [comment]
@DvidPiDev spam of `fixme:sync:NtCreateTransaction` in logs - happening only when you have 100% CPU usage bug.

When you launching game with no internet and see "error" message - you should not have any "log spam" - I mean if you had log spam - you did not make this "fix" work correctly.

**Comment by GMrlsN** [comment]
the nmcli command works well to mitigate the 100% CPU usage, but the game logs me out of my account every time I launch, is there anything I'm missing? or maybe another workaround?

**Comment by danilw** [comment]
@GMrlsN maybe you click error message when have no internet?

I launched game for last days with "no internet" - then saw error message - I did not click anything - turn on internet - click on "ingame message" when had internet - it auto login, no relog required.

**Comment by mrvictory1** [comment]
@GMrlsN try reducing "sleep 15" to "sleep 10" or play with different values, 15 is not a hard number

**Comment by GMrlsN** [comment]
Thanks for your help! I found that a 5 works well. I also encountered another issue: Steam couldn't close the game from its menu, so I adjusted the parameters. If anyone else faces the same problem, this worked for me:
nmcli n off ; %command% & pid=$! && sleep 5 ; nmcli n on && wait $pid

**Comment by Quackdoc** [comment]
is do we launch the launcher or the game directly?

also it is probably possible to use firejail to limit bandwidth to next to nothing, which could probably cause a timeout without being disruptive to anything else. I have yet to try this but it's something like this.

you may need to use `--noprofile`
```
firejail --name=limnet --net=eth0 STARTUPCMD
firejail --bandwidth=limnet set eth0 10 10 
<wait for timeout>
firejail --bandwidth=limnet set eth0 10000 10000
```

**Comment by 5310** [comment]
That's great news. Nmcli breaks my Tailscale Exits unkess I restart the services, I'd live to use to Firejail! 

(I only launch AAGL to update, and run the game directly through Steam to play. Still got the issue though.)

**Comment by mrvictory1** [comment]
issue persists as of 4.8 on my end

**Comment by Quackdoc** [comment]
Seems hit or miss for me, I use 3dmigoto, launching it sometimes will work fine, sometimes I need to use firejail

**Comment by Travitech** [comment]
Looking to me like this problem was resolved with the 5.0 update, tried starting it several times without any workarounds and MangoHud is showing expected cpu usage. Can anyone else confirm this on their end?

**Comment by 5310** [comment]
Gone for me too :+1:

**Comment by danilw** [comment]
Yes it solved _for now_ - and it unknown if in future they introduce it again, or as it was in 4.8 - it just appears week after again.

**Comment by mrvictory1** [comment]
@danilw on 4.8 the issue returned for me in mere 2 days.

**Comment by danilw** [comment]
I tested it yesterday just once with/without internet - had same CPU usage...

Now - today - I just started game again and CPU at 100% even in menu and map where it usually at 30%.

**I think this "bug" is back - and offline launch is working as solution, I tested.**

**Comment by mrvictory1** [comment]
I just launched an hour ago without workaround, CPU usage is normal on my end.

**Comment by danilw** [comment]
Few people confirmed same behavior - if you dont see high CPU usage on normal vs offline launch - then you dont need workaround, use workaround when you notice high CPU usage.

**Comment by mrvictory1** [comment]
@danilw I had this issue since it manifested in 4.7. Now I relaunched the game to test once more and it's gone. Which server are you on? (I'm on EU)

**Comment by Quackdoc** [comment]
The first couple boots were fine for me, but now every launch I do the issue is re-occuring

**Comment by NSArtem** [comment]
I can confirm, the same happened to me — the first several launches were fine (three? five?), but now the issue is back again.

On a side note: the same happens on a mac when running the game with Crossover. It's just on a mac it's easier to overlook because of the raw CPU power, the CPU load when launched with internet connection is around ~300% and 170% when launched without.

**Comment by maffinca69** [comment]
@Quackdoc Hi. Which command i must write to boot options? You write 3 line with new line, but steam deck not supported multiline command....
Thanks

**Comment by llyyr** [comment]
Is this an issue that happened after a game update or after wine update? If it's after a wine update then I'll go bisect the specific commit that caused it

**Comment by mrvictory1** [comment]
@llyyr it happened after game update, there isn't a known Wine build w/o this bug.

**Comment by Quackdoc** [comment]
firejail trick seems to no longer be working for me

**Comment by llyyr** [comment]
Seems to be fixed in 5.1 for me

**Comment by 5310** [comment]
> > firejail trick seems to no longer be working for me
> 
> Does cutting out Internet still work? 
Going offline until the lobby screen still works.

> Seems to be fixed in 5.1 for me
Every new patch this issue (and presumably, the anticheat itself) is off for a few days. For me, it was fine until yesterday, and now if I don't go offline while starting the game it's the same ~100% CPU usage.

**Comment by ShadowFlarer** [comment]
5.2 and the issue still persist, actually it feels worse now, tried the offline launch method and it worked for a day or two but not anymore unfortunately.

**Comment by 5310** [comment]
An observation just in case it helps us find how to stop this someday—

I was away on vacation for the entirety of last week, and this week I still don't get the 100% CPU issue, like how it doesn't happen for the first week or so after a version update. The Deck was shut down for the entire duration. 

Hope this sticks, or at least that the offline method continues to work for me afterwards.

**Comment by danilw** [comment]
> 5.2 and the issue still persist, actually it feels worse now, tried the offline launch method and it worked for a day or two but not anymore unfortunately.

As in few new [comments on reddit](https://www.reddit.com/r/linux_gaming/comments/1ddlfly/genshin_100_cpu_usage_since_today/?sort=new) - multiple people confirmed/noticed - offline launch not "always" work.

Offline launch still work but not always, idk pattern - sometime it work first try sometime does not.
(after 5.2 update now it like that)

**Comment by moriz1** [comment]
As of time of writing, we're well into v5.5, and at least on my system, the CPU workaround no longer seems to be necessary.

game.log no longer has the `fixme:sync:NtCreateTransaction` spam, and CPU usage appears normal.

**Comment by mrvictory1** [comment]
@moriz1 on which server do you play? On 5.4 I had CPU bug on Europe even though others playing on different servers who had the bug before 5.4 no longer did with 5.4.

**Comment by 5310** [comment]
I am on the EU server. I too haven't had this issue ccur since late November 2024. No `fixme:sync:NtCreateTransaction` for me as well, although I hadn't checked the logs until much later this year when I installed AAGL on a new machine.

**Comment by moriz1** [comment]
> [@moriz1](https://github.com/moriz1) on which server do you play? On 5.4 I had CPU bug on Europe even though others playing on different servers who had the bug before 5.4 no longer did with 5.4.

I'm on US server.

I'm also using the latest launcher version with support for wine-staging-tkg-wow64 10.5. I don't know if this makes any difference.

**Comment by K1NGSSTH** [comment]
Just passing by to say that i also not having the issue anymore, the game is running very well, i'm on US server btw.

**Comment by KuczaRacza** [comment]
Today i ran into same issue. I've  Wine-Staging-TkG 9.18  switching to GE-Proton=9.23 helped, but captcha didn't work finally GE-Proton=9.7  works

**Comment by danilw** [comment]
> Just passing by to say that i also not having the issue anymore, the game is running very well, i'm on US server btw.

for me - tested today - game dont even start when online (I click play in Launcher - no errors in logs - minute latter launcher back and game crashes)

only offline launch work from launcher

so I not sure what is happening

**Comment by vanruesc** [comment]
I'm still getting high CPU usage when starting the game normally, i.e. with an active internet connection. Haven't logged in for a month, just updated and logged in yesterday.

| Online | Offline |
|-|-|
| <img src="https://github.com/user-attachments/assets/b941c4e8-12b2-4a54-938e-efd24d546598" width="400"> | <img src="https://github.com/user-attachments/assets/6d3f2f5b-d87b-4324-bbe6-a5c6244b7733" width="400"> |

launcher: 3.13.1, game: 5.5.0, edition: Global, lutris 7.2-2, dxvk-2.6

**Comment by 5310** [comment]
Fascinating! My single sample confirmation that I have always been launching the game using Steam's Proton (9 stable now), never having used the Wine prefix that AAGL creates.

**Comment by danilw** [comment]
For me now - Proton (any version) does not work for launcher and game.

game can start with offline internet directly but it wont connect to internet when internet is ON

Only what working for me - wine 10.4 staging - and offline launch.

**Comment by danilw** [comment]
I close this - because this no longer relevant 
https://github.com/an-anime-team/an-anime-game-launcher/issues/483 - this is new one with game not launching

**Comment by krittinen** [comment]
100% bug seems to be affecting ZZZ as well.

**Comment by p0358** [comment]
With game version 5.6 it seems that my game's CPU usage have dropped from 800+% to <200%, perhaps the issue is now gone?

> anticheat mechanism now checks for "wine version" in Genshin and Zenles - they updated anticheat and "it know wine exist". There also integrated mechanism to "detect steamdeck exclusively". So they know.
(and do not contact their support mentioning about wine - it all unofficial)

Interesting, I wonder if it's rather for banning purposes or just for statistical to see how many people are playing on Linuxes actually. I always kept putting "add Linux support" in the surveys, even if using actual Windows

**Comment by yuusharo** [comment]
> With game version 5.6 it seems that my game's CPU usage have dropped from 800+% to <200%, perhaps the issue is now gone?

This happens every patch, the issue temporarily goes away for about a week or so, then returns like usual.

I question why this issue was marked closed, it is still very much still ongoing up through the start of 5.6 when it was marked closed, like will be again in a few days.

**Comment by danilw** [comment]
> I question why this issue was marked closed

I reopen it.

As for now - Proton 10 can run launcher and game (offline launch for game needed only to avoid hight CPu usage). And wine staging(10.4+) can run launcher but for game required offline launcher

**Comment by Quackdoc** [comment]
firejail method seems that it might be kinda broken now? for me the game will launch but hang right after the title fades out, (before the warning)

**Comment by Quackdoc** [comment]
@p0358 I use firejail to kill internet accsess, seems to be working again after I force reinstalled all my core apps. I don't currently use the launcher specifically on my main PC, so the command is a bit different, but it still works with the launcher on my laptop.

what I do is run

```
WINEPREFIX=/nvme/Bottles/Nvme-Games firejail --noprofile --name=limnet --net=wlan0  /home/quack/.local/share/bottles/runners/proton-tkg-nightly-8c0627b-07-10-2025/files/bin/wine64-tkg start /wait 'C:\Program Files\Genshin Impact\Genshin Impact game\GenshinImpact.exe'

firejail --bandwidth=limnet set wlan0 1 1

# wait for boot

firejail --bandwidth=limnet set wlan0 999999 999999
```

firejail will "block" the network accsess using the first firejail --bandwidth command by literally throttling it until it cant actually connect, the seccond command gives it a lot more bandwidth and allows it to work again.
